plugin/src/main/java/org/gradle/testretry/internal/executer/RetryTestResultProcessor.java

```diff
@@ -127,13 +133,24 @@ public void completed(Object testId, TestCompleteEvent testCompleteEvent) {
                         }
                     });
                 }
-
             }
         }
 
         delegate.completed(testId, testCompleteEvent);
     }
 
+    private void ignoreExpectedUnretriedTests() {
+        // check with the framework implementation if it is expected
+        Map<String, Set<String>> expectedUnretriedTests = previousRoundFailedTests.stream()
+            .collect(toMap(
+                Map.Entry::getKey,
+                entry -> entry.getValue().stream()
+                    .filter(test -> testFrameworkStrategy.isExpectedUnretriedTest(entry.getKey(), test))
+                    .collect(Collectors.toSet())
+            ));
+        expectedUnretriedTests.forEach((className, tests) -> previousRoundFailedTests.remove(className, tests::contains));
```

Comment:
💬 Ideally, we would also report failed descriptors as successful, but at this point, we have no clue where in the hierarchy they belong or how to construct a valid descriptor.

With all of the issues with the TestNG reporting, it seems like putting lipstick on a pig. So, I suggest not to bother and just make sure that retries work correctly.